vscode-lean4/src/infoview.ts

```diff
@@ -97,7 +102,18 @@ export class InfoProvider implements Disposable {
         sendClientRequest: async (uri: string, method: string, params: any): Promise<any> => {
             const client = this.clientProvider.findClient(uri);
             if (client) {
-                return client.sendRequest(method, params);
+                try {
+                    const result = await client.sendRequest(method, params);
+                    return result
+                } catch (ex) {
+                    if (ex.code === -32901 || ex.code === -32902) {
+                        // ex codes related with worker exited or crashed
+                        console.log(`The Lean Server has stopped processing this file: ${ex.message}`)
+                        await this.onWorkerStopped(uri, client, {message:'The Lean Server has stopped processing this file: ', reason: ex.message as string})
+                    } else {
+                        // file updated, do nothing
+                    }
```

Comment:
```suggestion
                    }
                    throw ex;
```
Exceptions are also used for "normal" control flow in the RPC interface (and the infoview code wants to get them to relay it to the user).

Note that this function here only intercepts some of the LSP request, namely those that are initiated by the infoview.  (Which should be enough since the infoview does lots of them while it's open.  I'm not even sure if we can intercept all LSP requests with the vscode-languageserver-node library.)